### DISCUSSION TITLE
Comparison between Guidance and AICI

### DISCUSSION THREAD

**Discussion by rwdr0** [guidance-ai/guidance#875] [Q&A]
I was wondering how Guidance and [Microsoft's Artificial Intelligence Controller Interface](https://github.com/microsoft/aici) are related? Both apparently claim to do the same thing.

**Comment by Harsha-Nori** [comment]
We're all part of the same team here and working together :), just thinking about each project as addressing different layers of the stack. AICI is meant to focus on lower level/cloud service layers, whereas guidance is more of a client facing library. We'll have more to say on how Guidance <> AICI work together shortly!

@mmoskal

**Reply by rwdr0** [reply]
Thank you for the answer. 

I did some digging around- and it seems that for now, AICI provides a greater degree of control over token generation than _Guidance_.

Take [AICI Backtracking](https://github.com/microsoft/aici/tree/main/controllers/pyctrl#backtracking) for example- I can already think of ways of using this as an on-the-fly error correction mechanism. 

Is there a way to do similar _Backtracking_ in Guidance?
